fe1-web/src/features/wallet/screens/WalletHome.tsx

```diff
@@ -1,51 +1,121 @@
 
-import { TextBlock, WideButtonView } from 'core/components';
+import { QRCode, TextBlock, WideButtonView } from 'core/components';
 import containerStyles from 'core/styles/stylesheets/containerStyles';
+import { LaoEventType } from 'features/events/objects';
+import { makeEventByTypeSelector } from 'features/events/reducer';
+import { selectCurrentLao } from 'features/lao/reducer';
+import { RollCall } from 'features/rollCall/objects';
+import PROPS_TYPE from 'resources/Props';
 import STRINGS from 'resources/strings';
 
-import { WalletStore } from '../store';
+import { RollCallTokensDropDown } from '../components';
+import * as Wallet from '../objects';
+import { createDummyWalletState, clearDummyWalletState } from '../objects/DummyWallet';
+import { RollCallToken } from '../objects/RollCallToken';
 
 const styles = StyleSheet.create({
+  homeContainer: {
+    ...containerStyles.centeredXY,
+    padding: '30px',
+  } as ViewStyle,
   smallPadding: {
     padding: '1rem',
   } as ViewStyle,
-  largePadding: {
-    padding: '2rem',
+  tokenSelectContainer: {
+    flexDirection: 'row',
+    marginTop: 30,
+    marginBottom: 20,
   } as ViewStyle,
 });
 
+const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
+
 /**
- * Wallet home screen
+ * Wallet UI once the wallet is synced
  */
-const WalletHome = () => {
-  // FIXME: Navigation should use a defined type here (instead of any)
-  const navigation = useNavigation<any>();
+const WalletHome = ({ navigation }: IPropTypes) => {
+  const [tokens, setTokens] = useState<RollCallToken[]>();
+  const [selectedToken, setSelectedToken] = useState<RollCallToken>();
+  const [isDebug, setIsDebug] = useState(false);
+  const rollCalls = useSelector(rollCallSelector);
+  const lao = useSelector(selectCurrentLao);
+
+  useEffect(() => {
+    if (lao && rollCalls) {
+      Wallet.recoverWalletRollCallTokens(rollCalls, lao)
+        .then((rct) => {
+          setTokens(rct);
+          setSelectedToken(rct[0]);
+        })
+        .catch((e) => {
+          console.debug(e);
+        });
+    } else {
+      // Clear tokens screen state
+      setSelectedToken(undefined);
+      setTokens(undefined);
+    }
+  }, [rollCalls, isDebug, lao]);
 
-  function importSeed() {
-    if (WalletStore.hasSeed()) {
-      navigation.navigate(STRINGS.navigation_synced_wallet);
+  const toggleDebugMode = () => {
+    if (isDebug) {
+      clearDummyWalletState();
+      setIsDebug(false);
     } else {
-      navigation.navigate(STRINGS.navigation_insert_seed_tab_wallet);
+      createDummyWalletState().then(() => setIsDebug(true));
+    }
+  };
+  const tokenInfos = () => {
+    if (selectedToken) {
+      const rollCallName = `Roll Call name: ${selectedToken.rollCallName.valueOf()}`;
+      return (
+        <View style={containerStyles.centeredXY}>
+          <TextBlock size={18} text={rollCallName} />
+          <QRCode value={selectedToken.token.publicKey.valueOf()} visibility />
```

Comment:
When testing this file make sure to mock the QRCode, the jest snapshot will be huge otherwise 😂